Library/Homebrew/dev-cmd/boneyard-formula-pr.rb

```diff
@@ -0,0 +1,125 @@
+    end
+    unless Formula["hub"].any_version_installed?
+      if ARGV.dry_run?
+        puts "brew install hub"
+      else
+        safe_system "brew", "install", "hub"
+      end
+    end
+    branch = "#{formula.name}-boneyard"
+    puts "cd #{formula.tap.formula_dir}" if ARGV.dry_run?
+    cd formula.tap.formula_dir
+    if ARGV.dry_run?
+      puts "git checkout -b #{branch} origin/master"
+      puts "git commit --no-edit --verbose --message=\"#{formula.name}: boneyard\" -- #{formula_path} #{formula.tap.path}/tap_migrations.json"
+      puts "hub fork --no-remote"
+      puts "hub fork"
+      puts "hub fork (to read $HUB_REMOTE)"
+      puts "git push $HUB_REMOTE #{branch}:#{branch}"
+      puts "hub pull-request -m $'#{formula.name}: boneyard\\n\\nCreated with `brew boneyard-formula-pr`.'"
+    else
+      safe_system "git", "checkout", "-b", branch, "origin/master"
+      safe_system "git", "commit", "--no-edit", "--verbose",
+        "--message=#{formula.name}: boneyard",
+        "--", formula_path, formula.tap.path/"tap_migrations.json"
+      safe_system "hub", "fork", "--no-remote"
+      quiet_system "hub", "fork"
+      remote = Utils.popen_read("hub fork 2>&1")[/fatal: remote (.+) already exists./, 1]
+      odie "cannot get remote from 'hub'!" if remote.to_s.empty?
+      safe_system "git", "push", remote, "#{branch}:#{branch}"
+      safe_system "hub", "pull-request", "-m",
+       "#{formula.name}: boneyard\n\nCreated with `brew boneyard-formula-pr`."
+    end
+    puts "cd #{Tap.fetch("homebrew", "boneyard").formula_dir}" if ARGV.dry_run?
+    cd Tap.fetch("homebrew", "boneyard").formula_dir
+    if ARGV.dry_run?
+      puts "git checkout -b #{branch} origin/master"
+      if bottle_block
+        puts "Removing bottle block"
+      else
+        puts "No bottle block to remove"
+      end
+      puts "git add #{formula.name}.rb"
+      puts "git commit --no-edit --verbose --message=\"#{formula.name}: boneyard\" -- #{formula.name}.rb"
+      puts "hub fork --no-remote"
+      puts "hub fork"
+      puts "hub fork (to read $HUB_REMOTE)"
+      puts "git push $HUB_REMOTE #{branch}:#{branch}"
+      puts "hub pull-request -m $'#{formula.name}: boneyard\\n\\nCreated with `brew boneyard-formula-pr`.'"
+    else
+      safe_system "git", "checkout", "-b", branch, "origin/master"
+      if bottle_block
+        Utils::Inreplace.inreplace("#{formula.name}.rb") do |s|
+          s.sub!(/  bottle do.+?end\n\n/m, "")
+        end
+      end
+      safe_system "git", "add", "#{formula.name}.rb"
+      safe_system "git", "commit", "--no-edit", "--verbose",
+        "--message=#{formula.name}: boneyard",
+        "--", "#{formula.name}.rb"
+      safe_system "hub", "fork", "--no-remote"
+      quiet_system "hub", "fork"
+      remote = Utils.popen_read("hub fork 2>&1")[/fatal: remote (.+) already exists./, 1]
+      odie "cannot get remote from 'hub'!" if remote.to_s.empty?
```

Comment:
You can simplify this to `unless remote` or `if remote.nil?`. If there is a match, it is guaranteed to be non-empty thanks to the `+` in the regular expression. If there is no match, `remote` will always be `nil`.